src/number/streaming.rs

```diff
@@ -1578,6 +1663,79 @@ where
   Ok((i, float))
 }
 
+/// Recognizes floating point number in text format and returns a f64.
+/// This only handles non-finite (special) values.
+pub fn double_nonfinite<T, E: ParseError<T>>(input: T) -> IResult<T, f64, E>
+where
+  T: Slice<RangeFrom<usize>> + Slice<RangeTo<usize>>,
+  T: Clone + Offset,
+  T: InputIter + InputLength + InputTake + crate::traits::ParseTo<i32>,
+  <T as InputIter>::Item: AsChar,
+  <T as InputIter>::IterElem: Clone,
+  T: InputTakeAtPosition,
+  <T as InputTakeAtPosition>::Item: AsChar,
+  T: AsBytes,
+  T: for<'a> Compare<&'a [u8]>,
+{
+  let (i, sign) = sign(input.clone())?;
+  let (mut double, count) = if case_insensitive_cmp(i.as_bytes(), b"nan") {
+    (f64::NAN, 3)
+  } else if i.input_len() >= 8 && case_insensitive_cmp(i.as_bytes(), b"infinity") {
+    (f64::INFINITY, 8)
+  } else if case_insensitive_cmp(i.as_bytes(), b"inf") {
```

Comment:
why this check after the `infinity` one ?

Reply:
Because if the string is `b"infinity"`, then `b"inf"` would also be true, so we must check the longer overload first.

Reply:
oh yes, ofcourse, maybe we could factorize this

Reply:
At the very least, maybe I should leave a comment?

Reply:
add a test would be better in my opinion

Reply:
This should be resolved now this this change:

https://github.com/Alexhuszagh/nom/blob/805e5be9d7e16a6fc0d86e3e6a16fd042d97fd3f/src/number/streaming.rs#L2185-L2190